src/realm/cluster.hpp

```diff
@@ -124,7 +124,10 @@ class ClusterNode : public Array {
     }
     int64_t get_key_value(size_t ndx) const
     {
-        return m_keys.get(ndx);
+        // BUG here: this is an unsigned array, if the array is compressed the value returned is not signed, so
+        // effectively, if the key is for example -3, the value returned here will be 11... (1) => sign bit (011) => 5
+        auto v = (int64_t)m_keys.get(ndx);
```

Comment:
This is breaking for compressed arrays, in particular for tombstones, instead of returning the negative value, we are returning an unsigned value, which is causing havoc, everywhere, casting to `int64_t` although sufficient to retrieve the right value is actually causing problems during lower bound, where values are compared unsigned for compressed arrays. TODO: find a better fix for this.